src/docfx/config/ops/OpsMetadataRuleConverter.cs

```diff
@@ -161,45 +162,71 @@ private static bool TryGetAttributeCustomRules(Dictionary<string, OpsMetadataRul
             return attributeCustomRules.Count != 0;
         }
 
-        private static bool TryGetAllowlist(
-            string parentName, string? listId, Allowlists allowlists, int index, out Dictionary<string, EnumDependenciesSchema?> allowList)
+        private static bool TryGetAllowlist(string attribute, string? listId, AllowLists allowlists, out Dictionary<string, EnumDependenciesSchema?> allowList)
         {
             allowList = new Dictionary<string, EnumDependenciesSchema?>();
 
-            if (!string.IsNullOrEmpty(listId) && allowlists.TryGetValue(listId, out var mainAllowlist))
+            if (!string.IsNullOrEmpty(listId) && allowlists.TryGetValue(listId.Substring("list:".Length), out var subAllowlist))
             {
-                if (string.IsNullOrEmpty(mainAllowlist.DependentAttribute))
+                if (string.IsNullOrEmpty(subAllowlist.NestedValue))
                 {
-                    allowList = mainAllowlist.Values.Keys.ToDictionary(validValue => validValue, validValue => (EnumDependenciesSchema?)null);
+                    allowList = subAllowlist.NestedTaxonomy.list.ToDictionary(x => x, x => (EnumDependenciesSchema?)null);
                     return true;
                 }
-
-                if (string.Equals(parentName, mainAllowlist.DependentAttribute))
+                else
                 {
-                    index++;
-                }
+                    var nestedValue = subAllowlist.NestedValue;
+                    var index = 0;
 
-                foreach (var (validValue, subListId) in mainAllowlist.Values)
-                {
-                    if (TryGetAllowlist(mainAllowlist.DependentAttribute, subListId, allowlists, index, out var subAllowList))
+                    if (string.Equals("slug", subAllowlist.NestedValue, StringComparison.OrdinalIgnoreCase))
                     {
-                        allowList.Add(validValue, new EnumDependenciesSchema() { { $"{mainAllowlist.DependentAttribute}[{index}]", subAllowList } });
+                        index = 1;
+                        nestedValue = attribute;
                     }
-                }
 
-                return true;
+                    // msService => ms.service, msSubService => ms.subservice
+                    var first = true;
+                    nestedValue = string.Concat(nestedValue.Select(
+                        (x, i) =>
+                        {
+                            if (i > 0 && char.IsUpper(x) && first)
+                            {
+                                first = false;
+                                return "." + x.ToString();
+                            }
+                            return x.ToString();
+                        })).ToLowerInvariant();
```

Comment:
Its efficiency is not ideal, extract to another method and use StringBuilder for this purpose?

Reply:
Great idea.